packages/@react-stately/table/src/utils.ts

```diff
@@ -0,0 +1,113 @@
+  if (!width) {
+    return 1;
+  } 
+  let match = width.match(/^(\d+)(?=fr$)/);
+      // if width is the incorrect format, just deafult it to a 1fr
+  if (!match) {
+    console.warn(`width: ${width} is not a supported format, width should be a number (ex. 150), percentage (ex. '50%') or fr unit (ex. '2fr')`, 
+        'defaulting to \'1fr\'');
+    return 1;
+  }
+  return parseInt(match[0], 10);
+}
+
+export function parseStaticWidth(width: number | string, tableWidth: number): number {
+  if (typeof width === 'string') {
+    let match = width.match(/^(\d+)(?=%$)/);
+    if (!match) {
+      throw new Error('Only percentages or numbers are supported for static column widths');
+    }
+    return tableWidth * (parseInt(match[0], 10) / 100);
+  }
+  return width;
+}
+    
+    
+export function getMaxWidth(maxWidth: number | string, tableWidth: number): number {
+  return maxWidth != null
+        ? parseStaticWidth(maxWidth, tableWidth)
+        : Infinity;
+}
+
+export function getMinWidth(minWidth: number | string, tableWidth: number): number {
+  return minWidth != null
+      ? parseStaticWidth(minWidth, tableWidth)
+      : 75;
+}
+
+function mapDynamicColumns<T>(remainingColumns: GridNode<T>[], remainingSpace: number, tableWidth: number): mappedColumn<T>[] {
+  let remainingFractions = remainingColumns.reduce(
+        (sum, column) => sum + parseFractionalUnit(column.props.defaultWidth),
+        0
+      );
+    
+  let columns = [...remainingColumns].map((column, index) => {
+    const targetWidth =
+          (parseFractionalUnit(column.props.defaultWidth) * remainingSpace) / remainingFractions;
+    
+    return {
+      ...column,
+      index,
+      delta: Math.max(
+            0,
+            getMinWidth(column.props.minWidth, tableWidth) - targetWidth,
+            targetWidth - getMaxWidth(column.props.maxWidth, tableWidth)
+          )
+    };
+  });
+    
+  return columns;
+}
+
+function findDynamicColumnWidths<T>(remainingColumns: mappedColumn<T>[], remainingSpace: number, tableWidth: number): mappedColumn<T>[] {
+  let remainingFractions = remainingColumns.reduce(
+        (sum, col) => sum + parseFractionalUnit(col.props.defaultWidth),
+        0
+      );
+    
+  for (let i = 0; i < remainingColumns.length; i++) {
+    const column = remainingColumns[i];
+    
+    const targetWidth =
+          (parseFractionalUnit(column.props.defaultWidth) * remainingSpace) / remainingFractions;
+    
+    let width = Math.max(
+          getMinWidth(column.props.minWidth, tableWidth),
+          Math.min(Math.floor(targetWidth), getMaxWidth(column.props.maxWidth, tableWidth))
+        );
+    column.calculatedWidth = width;
+    remainingSpace -= width;
+    remainingFractions -= parseFractionalUnit(column.props.defaultWidth);
+  }
+    
+  return remainingColumns;
+}  
+    
+export function getDynamicColumnWidths<T>(remainingColumns: GridNode<T>[], remainingSpace: number, tableWidth: number) {
```

Comment:
this isn't "remainingColumns" this is just all dynamic columns right?
and it's not "remainingSpace", it's "availableSpace" for the dynamic columns
is that correct? Or is this truly a subset of the dynamic columns and a subset of the available space for dynamic columns?

similar comments on some of the other functions in this file